### PR TITLE
[ASSIST-481] Message Center Email or SMS

### PR DESCRIPTION
**Pull request by dgoetzit**
This PR introduces the ability to "engage" Educatables through the Message Center's embedded timeline view. After selecting an educatable from the inbox, a user will be presented with the timeline view and an "Engage" action, which will initiate the Engagement creation wizard. On the Engagement "send", the Engagement is appended to the timeline. In the future, we will utilize event broadcasting to update the EngagementDeliverables status, but for now, and given how quickly everything processes, the status should be accurately reflected as soon as it is created.

This PR also resolves the outstanding pagination issues that existed in the Message Center previously.

### CODE REVIEW

File: app-modules/engagement/database/seeders/EngagementSeeder.php [hunk 1]
```diff
@@ -19,12 +19,12 @@ public function run(): void
...
-            ->has(EngagementDeliverable::factory()->count(1), 'engagementDeliverables')
-            ->forStudent()
-            ->deliverLater()
-            ->create();
+        // Engagement::factory()
+        //     ->count(7)
+        //     ->has(EngagementDeliverable::factory()->count(1), 'engagementDeliverables')
+        //     ->forStudent()
+        //     ->deliverLater()
+        //     ->create();
```
Comment (Orrison) [inline app-modules/engagement/database/seeders/EngagementSeeder.php:27]: Just curios the purpose of commenting these out here.

File: app-modules/engagement/src/Actions/DeliverEngagements.php [hunk 2]
```diff
@@ -17,10 +17,12 @@ class DeliverEngagements implements ShouldQueue
     use Queueable;
     use SerializesModels;
 
+    // TODO Add another indicator to the engagements table to represent an engagement being scheduled
+    // This will allow us to scope this query down further, and prevent any overlap with sync dispatches
```
Comment (Orrison) [inline app-modules/engagement/src/Actions/DeliverEngagements.php:21]: Let's just make sure we have a ticket for this TODO to ensure this gets worked on during Tech Debt burndown

File: app-modules/engagement/src/Notifications/EngagementEmailSentNotification.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            // TODO Remove reliance on "preferred" as a field - extract to attribute
+            ->line("Your engagement was successfully delivered to {$this->engagement->recipient->preferred}.");
+    }
+
+    public function toDatabase(User $notifiable): array
+    {
+        return FilamentNotification::make()
+            ->success()
+            ->title('Engagement Email Successfully Delivered')
+            ->body("Your engagement email was successfully delivered to {$this->engagement->recipient->display_name}.")
```
Comment (Orrison) [inline app-modules/engagement/src/Notifications/EngagementEmailSentNotification.php:39]: I noticed we use `display_name` here but in other areas we use `preferred` with a note to fix it later.

Should we just use display_name?

File: app-modules/engagement/src/Notifications/EngagementNotification.php [hunk 4]
```diff
@@ -25,6 +25,7 @@ public function toMail(object $notifiable): MailMessage
     {
         return (new MailMessage())
             ->subject($this->deliverable->engagement->subject)
+            // TODO Remove reliance on "preferred" as a field - extract to attribute
             ->greeting('Hello ' . $this->deliverable->engagement->recipient->preferred . '!')
```
Comment (Orrison) [inline app-modules/engagement/src/Notifications/EngagementNotification.php:29]: Same as the above display name question

File: app-modules/engagement/database/seeders/EngagementSeeder.php (hunk 1 above)
Comment (dgoetzit) [reply app-modules/engagement/database/seeders/EngagementSeeder.php:27]: I will add them back in - removed during testing.

File: app-modules/engagement/src/Actions/DeliverEngagements.php (hunk 2 above)
Comment (dgoetzit) [reply app-modules/engagement/src/Actions/DeliverEngagements.php:21]: Yep! Ticket here: https://engage.canyongbs.com/workgroups/group/3/tasks/task/view/601/

File: app-modules/engagement/src/Notifications/EngagementEmailSentNotification.php (hunk 3 above)
Comment (dgoetzit) [reply app-modules/engagement/src/Notifications/EngagementEmailSentNotification.php:39]: The only difference here is in who receives the notification. I _think_ (though I could be wrong) we'll want to use `preferred` when engaging with the Educatable, whereas admins might want to see `display_name`. But, we can probably discuss this strategy and what the right approach is.

Comment (Orrison) [reply app-modules/engagement/src/Notifications/EngagementEmailSentNotification.php:39]: Wouldn't both these notifications be going to an admin? Or do we we use toMail as like a response for an incoming Engagement?

**Review by Orrison — CHANGES_REQUESTED** [review]
Looks like tests are failing

**Review by Orrison — APPROVED** [review]

**Pull request merged**